scripts/event-generator-sheets.mjs

```diff
@@ -0,0 +1,169 @@
+import fs from 'fs';
+import dotenv from 'dotenv';
+import { google } from 'googleapis';
+import { getCssStringFromCommittee, generateSingleEvent } from './lib.mjs';
+
+// .env config
+dotenv.config();
+const SPREADSHEET_ID = process.env.EVENTS_SPREADSHEET_ID;
+const SERVICE_ACCOUNT = process.env.SERVICE_ACCOUNT ?? '';
+
+// Week one MONDAY of the quarter (y, m (base 0), d)
+const FIRST_DAY_OF_QUARTER = new Date(2023, 0, 9);
+const DAYS_OF_WEEK = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday'];
+
+// Grab all single and recurring events of Week n
+// and write to output.json
+async function writeAllEventsOfWeek(n) {
+  // Get events
+  let events = (await getSingleEventsOfWeek(n)).concat(await getRecurringEventsOfWeek(n));
+  const cleaned = events.filter((item) => item);
+  writeToOutput(cleaned);
+}
+
+// Get all single and recurring events of the quarter
+// Return as a list of JSONs
+async function getAllEvents() {
+  let events = [];
+
+  for (let i = 1; i <= 10; i++) {
+    events = events.concat(await getSingleEventsOfWeek(i));
+    events = events.concat(await getRecurringEventsOfWeek(i));
+  }
+
+  return events;
+}
+
+// Read single events of Week n
+// Return as array of JSON objects
+async function getSingleEventsOfWeek(n) {
+  const rows = await getGoogleSheetData('Week ' + n + '!A:H');
+
+  const events = [];
+  for (const row of rows) {
+    // Skip header rows and example event
+    if (row.length < 5 || row[0] === 'Committee' || row[0].includes('Example:')) {
+      continue;
+    }
+
+    try{
+      events.push(generateSingleEvent({
+        id: null,
+        title: row[1],
+        start: null,
+        end: null,
+        committee: getCssStringFromCommittee(row[0]),
+        location: row[5],
+        description: row[6],
```

Comment:
We can make this a lot safer by using [nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) which will check to see if the value on the left exists. If not, it supplies the value on the right, like so:
```js
description: row[6] ?? ''
```
Up to you if you would rather skip an event if no description is provided or if you want to display it with an empty description. 
(fyi the site crashed when description was undefined oop)